### PR TITLE
JS to output scholarships based on json

### PR DESCRIPTION
**Pull request by nicklee**
Javascript to output the JSON supplied by BE as a list of scholarships on the form page:

https://rca-development.herokuapp.com/scholarship-enquiry/

### CODE REVIEW

File: rca/static_src/javascript/components/scholarship-list.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        this.getData();
+        this.bindEvents();
+    }
+
+    getData() {
+        // Ideally this would be refactored to use fetch
+        const xmlhttp = new XMLHttpRequest();
+        xmlhttp.onreadystatechange = function getScholarships() {
+            if (this.readyState === 4 && this.status === 200) {
+                const dataLocation = document.getElementById('id_scholarships');
```
Comment (SimonDEvans) [inline rca/static_src/javascript/components/scholarship-list.js:20]: Could keep this with the other vars defined in the constructor.

File: rca/static_src/javascript/components/scholarship-list.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                const scholarships = JSON.parse(this.responseText);
+                const scholarshipsData = scholarships
+                    .map((item, index) => {
```
Comment (SimonDEvans) [inline rca/static_src/javascript/components/scholarship-list.js:23]: Massive nitpick, but backticks afford the ability to format the HTML so it's more readable. E.g something like this (untested):

```
.map((item, index) => {
    return `
        <li>
            <label for="id_scholarships_${index}">
                <input id="id_scholarships_${index}" type="checkbox" name="scholarships" value="${item.id}">
                ${item.title}
            </label>
        </li>`;
})
```

**Review by SimonDEvans — APPROVED** [review]
Would be nice to use `fetch()` as per your comment, all good otherwise 😄

File: rca/static_src/javascript/components/scholarship-list.js (hunk 2 above)
Comment (nicklee) [reply rca/static_src/javascript/components/scholarship-list.js:23]: This is loads better, updated

File: rca/static_src/javascript/components/scholarship-list.js (hunk 1 above)
Comment (nicklee) [reply rca/static_src/javascript/components/scholarship-list.js:20]: Agreed, but it doesn't seem to work, confused me…

**Pull request merged**